### DISCUSSION TITLE
ntpServer issue with offline Tasmota device

### DISCUSSION THREAD

**Discussion by BasingaPenny** [arendst/Tasmota#18776] [Q&A]
Hi everybody, 

i've some sonoff tasmota devices v 9.4.0. 
I've set all three ntpServer to a local address, two windows machines and my firewall which is also running ntp-service. 
The wifi network does not provide internet access. 
Unfortunately, the time is set to year 1970 and my firewall logs is saying, that the device try to access ntp service in the internet: 

```
192.168.1.205:43179 ==> eth1.4 ==> eth0 ==> 89.58.43.2:123
192.168.1.205:43421 ==> eth1.4 ==> eth0 => 144.76.0.164:123
```

When i switch the wifi to another with internet access, the time is set correct. 

Has someone any idea for that problem?
Thank u so mcuh. 
best regards
BP

**Comment by sfromis** [comment]
I have no trouble making Tasmota use one of my local NTP servers, instead of going for public servers on the Internet. Even if you configure NtpServer1,2,3, Tasmota may still fall back to default public NTP servers if access to the configured ones fails. One risk factor would be if your router setup "for security reasons" blocks clients from accessing each other, "client isolation".

**Comment by pkkrusty** [comment]
Sounds like maybe none of your NTP servers are configured correctly? I.e. their stratum is high enough that tasmota doesn't see them as valid. 

Or maybe your NTP settings in Tasmota aren't being saved correctly? What does it show you when you type in ntpserver1 to the console?

**Reply by BasingaPenny** [reply]
Thanks for your reply. 
"client isolation" wasn't active. I also though of this but it wasn't active. 

As i saw in the firewall, tasmota asked for diffrent NTP-server, diffrent IP-Addresses. Don't know if the "fallback" is a bunch of ntp-server ip addresses or domain-pool. in order to change a firewall-rule, would be good to know what the fallback is in datail. 

Also the ntpserver have been set correctly. SaveData = 1 (ON) was not changed and even after a restart console showed them again. 

I checked if the ntp-server are answering from a windows10 machine in the same wifi-network. 
I used this command: 
`w32tm /stripchart /computer:IP-ADDRESS-LOCAL-NTP-SERVER`

Not sure what the result of this command should look like. 
Mine was like:
```
#1 It is 02.06.2023 18:09:47
#2 18:09:47, Error: 0x800705b4
#3 18:09:50, Error: 0x800705b4
#4 18:09:53, Error: 0x800705b4
```

So I'm not sure if this is the reason why it did not work. First line at least gave the correct time. 
I finally solved it with opening the firewall-rule for ntp-services that the time can be managed through a remote ntp server. 

Thank you for your help.

**Reply by sfromis** [reply]
If you did not get your local NTP server to work with Tasmota, you did not get much of a "solution".... Tasmota can use such servers. NTP client code in Tasmota has been rewritten since your pretty old Tasmota version, but I'm not saying that this would solve it for you.

Change log says:
> Fallback NTP server from x.pool.ntp.org if no ntpservers are configured

Where x is a random number from 0 to 3. "No" basically means "no working".
